test/e2e/storage/persistent_volumes-local.go

```diff
@@ -1038,15 +1038,12 @@ func setupLocalVolumeBlockFs(config *localTestConfig, node *v1.Node) *localTestV
 	hostDir := filepath.Join(hostBase, testDirName)
 	createAndMapBlockLocalVolume(config, hostDir, node)
 	loopDev := getBlockLoopDev(config, hostDir, node)
-	// format and mount at hostDir
-	// give others rwx for read/write testing
-	cmd := fmt.Sprintf("sudo mkfs -t ext4 %s && sudo mount -t ext4 %s %s && sudo chmod o+rwx %s", loopDev, loopDev, hostDir, hostDir)
-	_, err := issueNodeCommandWithResult(config, cmd, node)
-	Expect(err).NotTo(HaveOccurred())
 	// Populate block volume with testFile containing testFileContent.
-	volume := setupWriteTestFile(hostDir, config, BlockFsLocalVolumeType, node)
-	volume.hostDir = hostDir
-	volume.loopDevDir = loopDev
+	volume := setupWriteTestFile(loopDev, config, BlockFsLocalVolumeType, node)
+	// we do this in order to set block device path to local PV spec path directly
```

Comment:
Instead of changing the existing test, I would add a new volume type.  So we basically have 3 cases for a block device:
* Path = block, VolumeMode = block
* Path = block, VolumeMode = fs
* Path = fs, VolumeMode = fs

Reply:
done